cryosparc/job.py

```diff
@@ -1519,7 +1537,8 @@ def save_output(self, name: str, dataset: Dataset, *, refresh: bool = True):
             >>> job.save_output("picked_particles", particles)
 
         """
-        url = f"/external/projects/{self.project_uid}/jobs/{self.uid}/outputs/{name}/dataset"
+
+        url = f"/external/projects/{self.project_uid}/jobs/{self.uid}/outputs/{urllib.parse.quote_plus(name)}/dataset"
```

Comment:
This was initially the only fix I tried to allow arbitrary characters, but this still results in an error when an output name had an encoded slash. So now we just totally forbid special characters in the name. We may lift this restriction from future versions, should we ever move away from Flask.